### PR TITLE
Fix pad to multiple of

### LINKED ISSUE #25729: Error with resize_token_embeddings and LLama model
### System Info

- `transformers` version: 4.32.0
- Platform: Linux-5.15.0-1038-gcp-x86_64-with-glibc2.31
- Python version: 3.10.7
- Huggingface_hub version: 0.16.4
- Safetensors version: 0.3.2
- Accelerate version: 0.21.0
- Accelerate config:    not found
- PyTorch version (GPU?): 2.0.1+cu117 (True)

### Who can help?

@ArthurZucker 

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

In the last version of the method `resize_token_embeddings`, the `pad_token_to_multiple=N` arg allows to increase the dimension to the first  int divisibile by N larger than `new_num_tokens`. 

However, in the method, the `config.vocab_size` is still updated to the initial [new_num_tokens](https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_utils.py#L1438).

For the LLama model, this leads to an error when the logits are flattened [here](https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py#L837). 

### Expected behavior

We should update the vocab_size to the new_num_tokens that accounts for the arg pad_token_to_multiple

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
Fixes #25729, `pad_to_multiple_of` did not overwrite the `model.config.vocab_size` properly.

### CODE REVIEW

**Comment by ArthurZucker** [comment]
Will fix the test, if model is not tied, we should not update the embedding, but that is also `True` for all models !

File: src/transformers/modeling_utils.py [hunk 1]
```diff
@@ -1432,8 +1432,8 @@ def resize_token_embeddings(
             return model_embeds
 
         # Update base model and current model config
-        self.config.vocab_size = new_num_tokens
-        self.vocab_size = new_num_tokens
+        self.config.vocab_size = model_embeds.weight.shape[0]
```
Comment (Taytay) [inline src/transformers/modeling_utils.py:1435]: In the line above, if the user doesn't pass a new length in, but DOES pass in a pad_to_multiple_of, the number of embeddings will still change, but this function does an early exit still. I think that this should change to still update the vocab_sizes in the case that the pad_to_multiple_of causes the embeddings to change, even if they didn't pass in a new_num_tokens.

Comment (ArthurZucker) [reply src/transformers/modeling_utils.py:1435]: This is not the intended use case. We could add this as a feature, but the goal is to pad when we add tokens, not just pad the embedding.

**Comment by ArthurZucker** [comment]
I'll try to merge this asap 😉 sorry for the trouble!

File: src/transformers/modeling_utils.py (hunk 1 above)
Comment (Taytay) [reply src/transformers/modeling_utils.py:1435]: Oh - I can see why that would not be the intended use case. I'm not using it that way either. I was just pointing out that this use of the function without that parameter would still cause the model to get into an invalid loadable state.

**Review by LysandreJik — APPROVED** [review]
Thanks @ArthurZucker
